scout/utils/acmg.py

```diff
@@ -192,11 +192,18 @@ def get_acmg(acmg_terms):
     bp_terms = []
     for term in acmg_terms:
         if term.endswith("_Strong"):
-            ps_terms.append(term)
```

Comment:
Nice! Looking at the entire function there is a long list of ifs and I wonder if it could be simplified?

ChatGPT suggests the following, but I haven't tried it:

```
# Create dictionaries to map suffixes and prefixes to corresponding lists
suffix_map = {
    "_Strong": {"P": ps_terms, "B": bs_terms},
    "_Moderate": {"P": pm_terms},
    "_Supporting": {"P": pp_terms, "B": bp_terms},
}

prefix_map = {
    "PVS": (lambda: setattr(self, 'pvs', True)),
    "PS": ps_terms.append,
    "PM": pm_terms.append,
    "PP": pp_terms.append,
    "BA": (lambda: setattr(self, 'ba', True)),
    "BS": bs_terms.append,
    "BP": bp_terms.append,
}

# Iterate through the terms
for term in acmg_terms:
    # Check suffixes first
    for suffix, prefix_dict in suffix_map.items():
        if term.endswith(suffix):
            for prefix, term_list in prefix_dict.items():
                if term.startswith(prefix):
                    term_list.append(term)
                    break
            break
    else:  # Only check prefixes if no suffix match
        for prefix, action in prefix_map.items():
            if term.startswith(prefix):
                action(term)
                break
```

Reply:
Mm, agree there are lots of if/elif. But dont you think this code is a little eh strained? It looks a bit like it is trying to be LISP. It would do slightly more comparisons, but without testing I would also say it is probably correct (would have to double check e.g. the PVS assignment). It would help a lot to change the pvs and ba to arrays, but then one would have to update a million tests..

Reply:
I like the prefix and suffix map - kind of approach, but feel free to use the suggestion or not. Itäs working after all!